packages/suite/src/utils/wallet/transactionUtils.ts

```diff
@@ -367,3 +368,124 @@ export const enhanceTransaction = (
+    const labels: { [label: string]: string[] } = {};
+    const { outputLabels } = accountMetadata;
+
+    Object.keys(outputLabels).forEach(txid => {
+        Object.values(outputLabels[txid]).forEach(label => {
+            if (!labels[label]) {
+                labels[label] = [];
+            }
+
+            labels[label].push(txid);
+        });
+    });
+
+    return labels;
+};
+
+const groupAddressesByLabel = (accountMetadata: AccountMetadata) => {
+    const labels: { [label: string]: string[] } = {};
+    const { addressLabels } = accountMetadata;
+
+    Object.keys(addressLabels).forEach(address => {
+        const label = addressLabels[address];
+
+        if (!labels[label]) {
+            labels[label] = [];
+        }
+
+        labels[label].push(address);
+    });
+
+    return labels;
+};
+
+export const searchTransactions = (
+    transactions: WalletAccountTransaction[],
+    accountMetadata: AccountMetadata,
+    search: string,
+) => {
+    if (search === '') {
+        return transactions;
+    }
+
+    // If it's an amount search (starting with < or > operator)
+    const amountOperator = ['<', '>'].find(k => search.startsWith(k));
+    if (amountOperator) {
+        const amount = new BigNumber(search.replace(amountOperator, ''));
+        switch (amountOperator) {
+            case '<':
+                return transactions.filter(t =>
+                    new BigNumber(t.amount).isLessThanOrEqualTo(amount),
+                );
+            case '>':
+                return transactions.filter(t =>
+                    new BigNumber(t.amount).isGreaterThanOrEqualTo(amount),
+                );
+            // no default
+        }
+    }
+
+    const txsToSearch: string[] = [];
+
+    // Find by output label
+    const txsForOutputLabels = groupTransactionsByLabel(accountMetadata);
+    const foundTxsForOutputLabel = Object.keys(txsForOutputLabels).flatMap(label => {
+        if (label.toLowerCase().includes(search.toLowerCase())) {
+            return txsForOutputLabels[label];
+        }
+
+        return [];
+    });
+    txsToSearch.push(...foundTxsForOutputLabel);
+
+    // Find by address label
+    const addressesForLabel = groupAddressesByLabel(accountMetadata);
+    const foundAddressesForLabel = Object.keys(addressesForLabel).flatMap(label => {
+        if (label.toLowerCase().includes(search.toLowerCase())) {
+            return addressesForLabel[label];
+        }
+
+        return [];
+    });
+
+    // Find by address
+    const txsForAddresses = groupTransactionIdsByAddress(transactions);
+    const foundTxsForAddress = Object.keys(txsForAddresses).flatMap(address => {
+        if (
+            address.toLowerCase().includes(search.toLowerCase()) ||
+            foundAddressesForLabel.includes(address)
+        ) {
+            return txsForAddresses[address];
+        }
+
+        return [];
+    });
+    txsToSearch.push(...foundTxsForAddress);
```

Comment:
search by date is missing (like in old wallet)

Reply:
Added in f1cfbca. Implementation is a bit different from the old wallet. In my implementation, an exact needs to be supplied in a yyyy-mm-mm format. I will also add support for searching dates with similar operators to the amount (<, >). I think it would also be useful to add an info tool tip that explains how to use the search.